frontend/ui/widget/textboxwidget.lua

```diff
@@ -144,6 +163,15 @@ function TextBoxWidget:_splitCharWidthList()
             idx = idx + 1
             -- FIXME: reuse newline entry
             self.vertical_string_list[ln+1] = {text = "", offset = idx, width = 0}
+        else
+            -- If next char is a space, discard it so it does not become
+            -- an ugly leading space on this next line
+            if idx <= size and self.char_width_list[idx].char == " " then
+                -- It's probably not needed to append it to last line
+                -- (not sure if that's ok with cursor stuff)
+                -- self.vertical_string_list[ln].text = self.vertical_string_list[ln].text .. " "
```

Comment:
I think it's fine to just ignore spaces on line break. Cursor uses `self.vertical_string_list` too so it should be fine. To enable cursor, just set `TextBoxWidget.editable` to `true`. I would just write a comment says `discard spaces on line break boundary`.